hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystem.java

```diff
@@ -355,6 +355,10 @@ public FSDataOutputStream create(final Path f, final FsPermission permission, fi
 
     if (prefixMode == PrefixMode.BLOB) {
       validatePathOrSubPathDoesNotExist(qualifiedPath, tracingContext);
+      Path parent = qualifiedPath.getParent();
+      if (parent != null && !parent.isRoot()) {
+        mkdirs(parent);
```

Comment:
We should have under:
```
try {
mkdirs(parent);
} catch(FileAlreadyExistsException ex){}
```

Why???

Lets say there are two threads:
t1:fs.create("/d1/file1");
t2:fs.create("/d1/file2"):
```
Now lets say t1 check d1 and found its not there, prepare to create it....; context switch happens...
t2 check d1 not there, prepare to create and create.... context switch happens
t1 tries to create d1 -> FileAlreadyExistException .... the whole create fails for t1.